### PR TITLE
Reland "Upgrade Gradle and AGP versions to 7.5/7.2 and migrate benchmarks+examples"

### PR DESCRIPTION
**Pull request by GaryQian**
Relands https://github.com/flutter/flutter/pull/108197

This version uses the gradle file generation script under `dev/tools/bin` for the benchmarks directory and updates lockfiles across the board.

### CODE REVIEW

**Comment by christopherfujino** [comment]
Looks like this PR downgrades some pub packages, was that intended?

**Comment by GaryQian** [comment]
No, thanks for the heads up, it made it in there due to outdated checkout. Needed to fetch latest master.

File: dev/benchmarks/platform_views_layout_hybrid_composition/android/buildscript-gradle.lockfile [hunk 1]
```diff
@@ -1,120 +0,0 @@
-# This is a Gradle generated file for dependency locking.
```
Comment (christopherfujino) [inline dev/benchmarks/platform_views_layout_hybrid_composition/android/buildscript-gradle.lockfile:1]: Why was this lockfile deleted?

Comment (GaryQian) [reply dev/benchmarks/platform_views_layout_hybrid_composition/android/buildscript-gradle.lockfile:1]: The script seems to have failed on this project, I fixed it and reran.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**